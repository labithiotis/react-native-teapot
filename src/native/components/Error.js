import React, { Component, PropTypes } from 'react';
import ReactNative from 'react-native';
import { colours } from '../styles';
import color from 'color';
import FAIcon from 'react-native-vector-icons/FontAwesome';

const { View, Text, StyleSheet } = ReactNative;

export default class Error extends Component {
  static propTypes = {
    error: PropTypes.shape({
      icon: PropTypes.string,
      title: PropTypes.string,
      message: PropTypes.string,
    }).isRequired,
  };

  render() {
    const {
      styles = {},
      error: {
        icon = 'exclamation-triangle',
        title = 'Error!',
        message = 'Whoops, something funky happened...',
      },
    } = this.props;
    return (
      <View style={[defaultStyles.container, styles.icon]}>
        <FAIcon style={[defaultStyles.icon, styles.icon]} name={icon} />
        <Text style={[defaultStyles.title, styles.title]}>
          {title}
        </Text>
        <Text style={[defaultStyles.text, styles.text]}>
          {message}
        </Text>
      </View>
    );
  }
}

const defaultStyles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    padding: 30,
  },
  icon: {
    color: colours.red,
    fontSize: 80,
    textAlign: 'center',
    marginBottom: 5,
    opacity: 0.4,
    backgroundColor: 'transparent',
  },
  title: {
    color: colours.red,
    fontWeight: 'bold',
    fontSize: 22,
    textAlign: 'center',
    marginBottom: 5,
  },
  text: {
    color: color(colours.red).darken(0.3).string(),
    fontSize: 14,
    textAlign: 'center',
  },
});
