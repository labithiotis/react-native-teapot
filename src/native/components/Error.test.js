import React from 'react';
import { shallow } from 'enzyme';
import Error from './Error';

describe('Error', () => {
  it('should render correctly', () => {
    const component = shallow(
      <Error error={{ title: 'test', message: 'test' }} />,
    );
    expect(component).toMatchSnapshot();
  });
});
