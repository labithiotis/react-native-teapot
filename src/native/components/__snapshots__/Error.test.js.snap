// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 30,
      },
      undefined,
    ]
  }
>
  <Icon
    allowFontScaling={false}
    name="exclamation-triangle"
    size={12}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#F41C04",
          "fontSize": 80,
          "marginBottom": 5,
          "opacity": 0.4,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#F41C04",
          "fontSize": 22,
          "fontWeight": "bold",
          "marginBottom": 5,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    test
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "hsl(6, 96.8%, 34%)",
          "fontSize": 14,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    test
  </Text>
</View>
`;
